artillery-core/src/epidemic/state.rs

```diff
@@ -309,6 +312,19 @@ impl ArtilleryState {
                 let myself = self.members.leave();
                 enqueue_state_change(&mut self.state_changes, &[myself]);
             }
+            Payload(id, msg) => {
+                if let Some(target_peer) = self.members.get_member(&id) {
+                    self.process_request(TargetedRequest {
+                        request: Request::Payload(id, msg),
+                        target: target_peer.remote_host().unwrap(), // Does this panic?
+                    });
```

Comment:
Yes it might fail, Mallory can:
* sniff the network
* craft a packet with one of the node ids
* put malicious payload
* crash custom event routine of the destined node.

This remote host selection should be written with a defensive style like:
```suggestion
                if let Some(target_peer) = self.members.get_member(&id) {
                    if target_peer.is_remote() {
                        self.process_request(TargetedRequest {
                            request: Request::Payload(id, msg),
                            target: target_peer.remote_host().unwrap(),
                        });
                    } else {
                        error!("Current node can't send payload to self over LAN")
                    }
                }
```